### PR TITLE
[com_finder] - fix update inner join syntax for postgres

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
fix update join syntax for postgresql

### Testing Instructions

- with Content - Smart Search  plugin enabled
- index content
- unpublish/publish a category

### Expected result
no sql error

### Actual result
screenshot from 2018-03-22 08-07-52

### CODE REVIEW

File: administrator/components/com_finder/helpers/indexer/indexer.php [hunk 1]
```diff
@@ -306,10 +306,10 @@ public function remove($linkId)
 		{
 			// Update the link counts for the terms.
 			$query->clear()
-				->update($db->quoteName('#__finder_terms', 't'))
```
Comment (mbabker) [inline administrator/components/com_finder/helpers/indexer/indexer.php]: Can you please revert the changes to use the shortcuts?  When reading the code the full method names are clearer than the shortcuts, especially to newer contributors (IIRC at one point we tried standardizing the repo to use the full names over the shortcuts, or maybe I'm mis-remembering).

Comment (alikon) [reply administrator/components/com_finder/helpers/indexer/indexer.php]: readability  vs laziness, readability should win ;)
done

**Comment by csthomas** [comment]
I tested this PR with success but @alikon you have to add a code style fix:

```
FILE: ...mla-cms/administrator/components/com_finder/helpers/indexer/indexer.php
--------------------------------------------------------------------------------
FOUND 0 ERROR(S) AND 1 WARNING(S) AFFECTING 1 LINE(S)
--------------------------------------------------------------------------------
 310 | WARNING | Line exceeds 150 characters; contains 155 characters
--------------------------------------------------------------------------------
UPGRADE TO PHP_CODESNIFFER 2.0 TO FIX ERRORS AUTOMATICALLY
```

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on eabc72832015414e013b40af3a5b69c0dee7a552<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19964">issues.joomla.org/tracker/joomla-cms/19964</a>.</sub>

**Comment by GarryMoveOut** [comment]
I have tested this item :white_check_mark: successfully on eabc72832015414e013b40af3a5b69c0dee7a552<br /><br />Works for me. Everything is ok. Error disappear. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19964">issues.joomla.org/tracker/joomla-cms/19964</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.
